### PR TITLE
JAVA-30185 Upgrade mockito library in main pom.xml

### PR DESCRIPTION
**Pull request by anuragkumawat**

### CODE REVIEW

**Review by alvin577austria — APPROVED** [review]

File: core-java-modules/core-java-os/pom.xml [hunk 1]
```diff
@@ -67,6 +67,7 @@
         <collections-generic.version>4.01</collections-generic.version>
         <unix4j.version>0.4</unix4j.version>
         <grep4j.version>1.8.7</grep4j.version>
+        <mockito.version>4.4.0</mockito.version>
```
Comment (alvin577austria) [inline core-java-modules/core-java-os/pom.xml:70]: Why do we need to specify the version here?

File: libraries-http-2/pom.xml [hunk 2]
```diff
@@ -120,6 +120,7 @@
         <reactive.stream.version>1.0.3</reactive.stream.version>
         <reactor.version>3.6.0</reactor.version>
         <jmockit.version>1.49</jmockit.version>
+        <mockito.version>5.2.0</mockito.version>
```
Comment (alvin577austria) [inline libraries-http-2/pom.xml:123]: Shouldn't this be 5.11.0?

File: patterns-modules/design-patterns-creational-2/pom.xml [hunk 3]
```diff
@@ -20,5 +20,9 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+    
+    <properties>
+        <mockito.version>5.2.0</mockito.version>
```
Comment (alvin577austria) [inline patterns-modules/design-patterns-creational-2/pom.xml:25]: 5.11.0?

File: persistence-modules/spring-data-dynamodb/pom.xml [hunk 4]
```diff
@@ -174,6 +174,7 @@
         <dynamodblocal.version>1.21.1</dynamodblocal.version>
         <maven-dependency-plugin.version>3.1.1</maven-dependency-plugin.version>
         <spring-boot.version>2.4.7</spring-boot.version>
+        <mockito.version>4.4.0</mockito.version>
```
Comment (alvin577austria) [inline persistence-modules/spring-data-dynamodb/pom.xml:177]: Does 5.11.0 give failure?

File: spring-boot-modules/spring-boot-custom-starter/greeter-spring-boot-autoconfigure/pom.xml [hunk 5]
```diff
@@ -61,6 +61,7 @@
     <properties>
         <spring-boot.version>2.2.6.RELEASE</spring-boot.version>
         <greeter.version>0.0.1-SNAPSHOT</greeter.version>
+        <mockito.version>4.4.0</mockito.version>
```
Comment (alvin577austria) [inline spring-boot-modules/spring-boot-custom-starter/greeter-spring-boot-autoconfigure/pom.xml:64]: 5.11.0?

File: spring-security-modules/spring-security-legacy-oidc/pom.xml [hunk 6]
```diff
@@ -49,6 +49,7 @@
         <spring-security-jwt.version>1.0.9.RELEASE</spring-security-jwt.version>
         <jwks-rsa.version>0.3.0</jwks-rsa.version>
         <spring-boot.version>2.4.7</spring-boot.version>
+        <mockito.version>4.4.0</mockito.version>
```
Comment (alvin577austria) [inline spring-security-modules/spring-security-legacy-oidc/pom.xml:52]: Does 5.11.0 give an error?

**Review by alvin577austria — CHANGES_REQUESTED** [review]

File: spring-security-modules/spring-security-legacy-oidc/pom.xml (hunk 6 above)
Comment (anuragkumawat) [reply spring-security-modules/spring-security-legacy-oidc/pom.xml:52]: Yes, this is a quick fix to run the build. Once the build passes we can investigate the code changes required with the new version.

File: spring-boot-modules/spring-boot-custom-starter/greeter-spring-boot-autoconfigure/pom.xml (hunk 5 above)
Comment (anuragkumawat) [reply spring-boot-modules/spring-boot-custom-starter/greeter-spring-boot-autoconfigure/pom.xml:64]: Yes, this is a quick fix to run the build. Once the build passes we can investigate the code changes required with the new version.

File: persistence-modules/spring-data-dynamodb/pom.xml (hunk 4 above)
Comment (anuragkumawat) [reply persistence-modules/spring-data-dynamodb/pom.xml:177]: Yes, this is a quick fix to run the build. Once the build passes we can investigate the code changes required with the new version.

File: patterns-modules/design-patterns-creational-2/pom.xml (hunk 3 above)
Comment (anuragkumawat) [reply patterns-modules/design-patterns-creational-2/pom.xml]: The latest version for [mockito-inline](https://mvnrepository.com/artifact/org.mockito/mockito-inline/5.2.0) is 5.2.0, hence the version has been forced in some child modules.

File: libraries-http-2/pom.xml (hunk 2 above)
Comment (anuragkumawat) [reply libraries-http-2/pom.xml]: The latest version for [mockito-inline](https://mvnrepository.com/artifact/org.mockito/mockito-inline/5.2.0) is 5.2.0, hence the version has been forced in some child modules.

File: core-java-modules/core-java-os/pom.xml (hunk 1 above)
Comment (anuragkumawat) [reply core-java-modules/core-java-os/pom.xml:70]: Yes, this is a quick fix to run the build. Once the build passes we can investigate the code changes required with the new version.

Comment (alvin577austria) [reply core-java-modules/core-java-os/pom.xml:70]: OK, I will create separate task for this module.

File: graphql-modules/graphql-spqr-boot-starter/pom.xml [hunk 7]
```diff
@@ -36,6 +36,7 @@
 
     <properties>
         <graphql-spqr-spring-boot-starter-version>0.0.6</graphql-spqr-spring-boot-starter-version>
+        <mockito.version>4.4.0</mockito.version>
```
Comment (alvin577austria) [inline graphql-modules/graphql-spqr-boot-starter/pom.xml:39]: I will create a separate task for this module.

File: graphql-modules/graphql-spqr/pom.xml [hunk 8]
```diff
@@ -37,6 +37,7 @@
 
     <properties>
         <spqr-version>0.11.2</spqr-version>
+        <mockito.version>4.4.0</mockito.version>
```
Comment (alvin577austria) [inline graphql-modules/graphql-spqr/pom.xml:40]: Same as above

File: libraries-http-2/pom.xml [hunk 9]
```diff
@@ -120,6 +120,7 @@
         <reactive.stream.version>1.0.3</reactive.stream.version>
         <reactor.version>3.6.0</reactor.version>
         <jmockit.version>1.49</jmockit.version>
+        <mockito-inline.version>5.2.0</mockito-inline.version>
```
Comment (alvin577austria) [inline libraries-http-2/pom.xml:123]: Can you put this in the main pom.xml please.

File: patterns-modules/design-patterns-creational-2/pom.xml [hunk 10]
```diff
@@ -16,9 +16,13 @@
...
             <groupId>org.mockito</groupId>
             <artifactId>mockito-inline</artifactId>
-            <version>${mockito.version}</version>
+            <version>${mockito-inline.version}</version>
             <scope>test</scope>
         </dependency>
     </dependencies>
+    
+    <properties>
+        <mockito-inline.version>5.2.0</mockito-inline.version>
```
Comment (alvin577austria) [inline patterns-modules/design-patterns-creational-2/pom.xml:25]: Same here if possible please put in the main pom.xml

File: persistence-modules/spring-data-dynamodb/pom.xml (hunk 4 above)
Comment (alvin577austria) [reply persistence-modules/spring-data-dynamodb/pom.xml:177]: OK, I will create a separate task for this module.

File: spring-boot-modules/spring-boot-custom-starter/greeter-spring-boot-autoconfigure/pom.xml (hunk 5 above)
Comment (alvin577austria) [reply spring-boot-modules/spring-boot-custom-starter/greeter-spring-boot-autoconfigure/pom.xml:64]: OK, I will create a separate task for this module.

File: spring-security-modules/spring-security-legacy-oidc/pom.xml (hunk 6 above)
Comment (alvin577austria) [reply spring-security-modules/spring-security-legacy-oidc/pom.xml:52]: OK, I will create a separate task for this module.

File: testing-modules/mockito-2/pom.xml [hunk 11]
```diff
@@ -27,8 +27,7 @@
     </dependencies>
 
     <properties>
-        <mockito-inline.version>4.8.1</mockito-inline.version>
-        <mockito.version>5.9.0</mockito.version>
+        <mockito-inline.version>5.2.0</mockito-inline.version>
```
Comment (alvin577austria) [inline testing-modules/mockito-2/pom.xml:30]: Can we put this is the parent pom please.

**Review by alvin577austria — CHANGES_REQUESTED** [review]

File: testing-modules/mockito-2/pom.xml (hunk 11 above)
Comment (anuragkumawat) [reply testing-modules/mockito-2/pom.xml]: changes done

File: patterns-modules/design-patterns-creational-2/pom.xml (hunk 10 above)
Comment (anuragkumawat) [reply patterns-modules/design-patterns-creational-2/pom.xml]: changes done

File: libraries-http-2/pom.xml (hunk 9 above)
Comment (anuragkumawat) [reply libraries-http-2/pom.xml]: changes done

File: core-java-modules/core-java-os/pom.xml (hunk 1 above)
Comment (alvin577austria) [reply core-java-modules/core-java-os/pom.xml:70]: https://team.baeldung.com/browse/JAVA-33816

File: persistence-modules/spring-data-dynamodb/pom.xml (hunk 4 above)
Comment (alvin577austria) [reply persistence-modules/spring-data-dynamodb/pom.xml:177]: https://team.baeldung.com/browse/JAVA-33817

File: spring-boot-modules/spring-boot-custom-starter/greeter-spring-boot-autoconfigure/pom.xml (hunk 5 above)
Comment (alvin577austria) [reply spring-boot-modules/spring-boot-custom-starter/greeter-spring-boot-autoconfigure/pom.xml:64]: https://team.baeldung.com/browse/JAVA-33818

File: spring-security-modules/spring-security-legacy-oidc/pom.xml (hunk 6 above)
Comment (alvin577austria) [reply spring-security-modules/spring-security-legacy-oidc/pom.xml:52]: https://team.baeldung.com/browse/JAVA-33819

File: graphql-modules/graphql-spqr-boot-starter/pom.xml (hunk 7 above)
Comment (alvin577austria) [reply graphql-modules/graphql-spqr-boot-starter/pom.xml:39]: https://team.baeldung.com/browse/JAVA-33820

File: graphql-modules/graphql-spqr/pom.xml (hunk 8 above)
Comment (alvin577austria) [reply graphql-modules/graphql-spqr/pom.xml:40]: https://team.baeldung.com/browse/JAVA-33821

**Review by alvin577austria — APPROVED** [review]

**Pull request merged**